src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -716,35 +727,37 @@ public static void Fill<T>(ref T refData, nuint numElements, T value)
                 {
                     length -= 8;
 
-                    if (value.Equals(Unsafe.Add(ref searchSpace, length + 7)))
+                    ref T current = ref Unsafe.Add (ref searchSpace, length);
+                    if (value.Equals(Unsafe.Add(ref current, 7)))
                         goto Found7;
-                    if (value.Equals(Unsafe.Add(ref searchSpace, length + 6)))
+                    if (value.Equals(Unsafe.Add(ref current, 6)))
                         goto Found6;
-                    if (value.Equals(Unsafe.Add(ref searchSpace, length + 5)))
+                    if (value.Equals(Unsafe.Add(ref current, 5)))
                         goto Found5;
-                    if (value.Equals(Unsafe.Add(ref searchSpace, length + 4)))
+                    if (value.Equals(Unsafe.Add(ref current, 4)))
                         goto Found4;
-                    if (value.Equals(Unsafe.Add(ref searchSpace, length + 3)))
+                    if (value.Equals(Unsafe.Add(ref current, 3)))
                         goto Found3;
-                    if (value.Equals(Unsafe.Add(ref searchSpace, length + 2)))
+                    if (value.Equals(Unsafe.Add(ref current, 2)))
                         goto Found2;
-                    if (value.Equals(Unsafe.Add(ref searchSpace, length + 1)))
+                    if (value.Equals(Unsafe.Add(ref current, 1)))
                         goto Found1;
-                    if (value.Equals(Unsafe.Add(ref searchSpace, length)))
+                    if (value.Equals(current))
```

Comment:
I assume there is a difference between CLR and MonoVM which required you to apply this pattern to all the helper methods here. Do we have an issue that tracks it?